File: Repositories/Items/ToDo.cs
namespace Repositories.Items
{
    public class ToDo
    {
        public int Id { get; set; }
        public string Text { get; set; }
        public string? Description { get; set; }
        public bool IsComplete { get; set; } = false;
        public DateTime? Deadline { get; set; }
        public int? CategoryId { get; set; }
    }
}

File: client/src/types/toDo.ts
export interface ToDo{
    id?: number | string,
    text: string,
    description?: string,
    deadline?: string,
    isComplete: boolean,
    categoryId?: number | string
}

export enum ActionTypes{
    FETCH_TODOS = "FETCH_TODOS",
    FETCH_TODOS_SUCCESS = "FETCH_TODOS_SUCCESS",
    FETCH_TODOS_ERROR = "FETCH_TODOS_ERROR"
}

interface FetchToDosAction{type: ActionTypes.FETCH_TODOS}
interface FetchToDosSuccessAction{type: ActionTypes.FETCH_TODOS_SUCCESS, payload: ToDo[]}
interface FetchToDosErrorAction{type: ActionTypes.FETCH_TODOS_ERROR, payload: string}

export type ToDoAction = (FetchToDosAction | FetchToDosErrorAction | FetchToDosSuccessAction)

export interface ToDoState{
    toDoList: ToDo[],
    loading: boolean,
    error: null | string
}
